### PR TITLE
Fix flaky scale writers test

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 1]
```diff
@@ -1646,7 +1644,7 @@ public void testBucketedExecution()
                 "SELECT custkey, COUNT(*) FROM orders GROUP BY custkey");
     }
 
-    @Test
+    @Test(invocationCount = 100)
```
Comment (electrum) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: This is for testing in Travis. Will remove before merging.

**Comment by electrum** [comment]
Let me try decreasing the minimum size. With `sf2`, the output size of RCFiles is about 75MB. I picked `4MB` originally due to the RCFile buffer size, but we might as well make it as small as possible to encourage scaling as quickly as possible.

**Comment by electrum** [comment]
This passed in Travis with `@Test(invocationCount = 100)` (the job failed due to runtime exceeded but none of the `testScaleWriters` failed)

**Comment by findepi** [comment]
I run the new version (lowering `writer_min_size`) on our Jenkins with invocation count 20. Failed 19 of 20 times (always with the same message), while all other tests are passing 😕

**Comment by sopel39** [comment]
@electrum What are conditions for triggering writers scale up? Apparently those are still not met on our Jenkins machines. Test themselves take 2x time on our machines than on Travis (even though those are similar EC2 instances).

**Comment by electrum** [comment]
The conditions are

* `writer_min_size` times the number of writer tasks has been written in total (in other words, each writer has written this amount on average). This is the value reported by `ConnectorPageSink.getCompletedBytes()`, which for Hive writers is the amount written to the file system (so you need to wait for the file format to flush).
* At least 50% of the source tasks have the `outputBufferOverutilized` state, which means that the task is blocking producers. This is true when the buffer is at least 50% utilized or when "no more pages" has been set (meaning there will be no additional input).

You can see the (rather trivial) logic in `ScaledWriterScheduler` and `ArbitraryOutputBuffer`. The scheduling check happens every 200ms.

**Comment by electrum** [comment]
I find it strange that the failure is always 3 of 4.

**Comment by electrum** [comment]
Given this seems to improve Travis, do you think we should merge this now, and follow up on fixing it for your internal Jenkins?

**Comment by findepi** [comment]
> Maybe the data isn't coming from tpch connector fast enough in our env?

We could copy data first into memory connector and only then CTAS into hive.

**Comment by electrum** [comment]
After discussion with @dain, I updated it to check that the value is in the range of `[2, nodeCount]`. This gets it working for now, though it makes the test a bit weaker.

**Comment by sopel39** [comment]
> This is true when the buffer is at least 50% utilized or when "no more pages" has been set (meaning there will be no additional input).

@electrum why buffer with "no more pages" set is considered `outputBufferOverutilized`?

**Pull request merged**